File: models/Gpt-3.5-turbo/py/code/graph_algorithms/School_Dance.py

# Read input
n, m, k = map(int, input().split())

# Initialize arrays to keep track of availability
boys = [True] * n
girls = [True] * m

# Initialize an array to store the dance pairs
pairs = []

# Iterate through all potential pairs
for _ in range(k):
    a, b = map(int, input().split())
    
    # Check if the boy and girl are available
    if boys[a-1] and girls[b-1]:
        # Mark the boy and girl as unavailable
        boys[a-1] = False
        girls[b-1] = False
        
        # Add the pair to the list of pairs
        pairs.append((a, b))

# Print the maximum number of dance pairs
print(len(pairs))

# Print the pairs
for pair in pairs:
    print(pair[0], pair[1])

File: models/Gpt-3.5-turbo/cpp/code/problems/graph_algorithms/School_Dance.cpp

#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n, m, k;
    cin >> n >> m >> k;

    vector<bool> boy_available(n + 1, true); // to keep track of available boys
    vector<bool> girl_available(m + 1, true); // to keep track of available girls

    vector<pair<int, int>> dance_pairs; // to store the dance pairs

    for (int i = 0; i < k; i++) {
        int a, b;
        cin >> a >> b;

        // If both the boy and girl are available, form a pair
        if (boy_available[a] && girl_available[b]) {
            dance_pairs.push_back({a, b});
            boy_available[a] = false;
            girl_available[b] = false;
        }
    }

    int r = dance_pairs.size(); // the maximum number of dance pairs

    cout << r << endl;
    for (auto pair : dance_pairs) {
        cout << pair.first << " " << pair.second << endl;
    }

    return 0;
}
